daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala

```diff
@@ -143,7 +139,7 @@ object SValue {
       extends SValue
 
   @SuppressWarnings(Array("org.wartremover.warts.ArrayEquals"))
-  final case class SStruct(fields: ImmArray[Name], values: util.ArrayList[SValue]) extends SValue
+  final case class SStruct(fieldNames: Struct[Unit], values: util.ArrayList[SValue]) extends SValue
```

Comment:
Worth documenting that `values` must be orderd by name just like `fieldNames`?

Reply:
done